### PR TITLE
Refactor the planning/cms page as per #181

### PR DESCRIPTION
**Pull request by MikeyMJCO**
This is a PR to deal with #181 and #33.

### CODE REVIEW

File: docs/planning/cms.md [hunk 1]
```diff
@@ -1,61 +1,45 @@
...
+2. What information, collected through our website, do we want to store in CiviCRM (e.g. Family data from a public form with the family relationships automatically added).
+
+This is known as integration. How much integration you require will help you determine which is the best CMS for you.
+
+## Availability of Extensions
+
+There are historical reasons why different CMS options have different levels of integration with CiviCRM. CiviCRM version 1.x did not originally support its own extension mechanism - the only option was to write an integration in the form of a Drupal module. Consequently, CiviCRM Drupal integrations grew first and fastest. In CiviCRM version 4.2 a flexible, cross-CMS extension system was introduced to CiviCRM further reducing disparity between the CMS options.
+
+You may also choose a CMS because you already know how to use it or it is already used by your organisation. Each CMS has different advantages and disadvantages and different people will give you views on which is better. This chapter is aimed at giving you the requisite information to choose the best CMS for you.
```
Comment (mlutfy) [inline docs/planning/cms.md]: I would say that this is the number 1 reason to chose a CMS: familiarity. The other arguments do not stand, in my opinion. We encourage cross-CMS solutions, and CMS-specific extensions, while useful for power-users, are not something we should encourage.

Comment (MikeyMJCO) [reply docs/planning/cms.md]: Any suggested language change you have in mind? I agree that CiviCRM's position should be that the CMS choice is entirely for them to make and we'll work as hard as we can to support the CMS they choose - whilst being realistic about the differences as they stand presently.

File: docs/planning/cms.md [hunk 2]
```diff
@@ -1,61 +1,45 @@
...
-Individuals and organisations with a CiviCRM record are **Contacts** 
+## Integration Features
 
+Broadly there are some differences which arise because of distinctive features/norms in each CMS community, e.g.
 
-<sup><a name="1">1</a></sup> Usage of content management systems for websites, [Link](https://w3techs.com/technologies/overview/content_management/all).
- 
+* In WordPress, it's pretty common to embed content in your web-pages using short-codes. CiviCRM has several short-codes in WP.
+* In Backdrop CMS/Drupal 7/Drupal 8, it's pretty common to embed content in your web-pages using blocks. CiviCRM has several blocks in Drupal.
+* In Backdrop CMS/Drupal 7/Drupal 8, the core CMS has decent support for "roles" and "permissions", so CiviCRM uses this. In Joomla/WordPress, these are regarded more as "hidden/advanced" functionality, so CiviCRM has to fill in some gaps.
```
Comment (mlutfy) [inline docs/planning/cms.md]: Regarding "hidden" functionality in Joomla/WP, I'm not too familiar, but I feel like it should be articulated a bit more?

Also, Contrib/Event forms cannot be embedded as blocks in Drupal/Backdrop (I look forward to Gutenberg blocks!). CiviCRM has a few Drupal blocks, they are not useful for the website content itself. However, Views vs Content Views is probably still an important factor?

Ex: generating member listings, maps of CRM data.. easier in Drupal than WP.

Maybe this is too opinionated, but I would argue: WordPress has a lower barrier to adoption, easier to maintain; Drupal/Backdrop have a few extra features for power-users?

File: docs/planning/cms.md [hunk 3]
```diff
@@ -1,61 +1,45 @@
...
 
+## Installation Process
+
+The installers are a bit different - web-based installation is easier in Joomla, but it's more flexible in Drupal or WordPress. Command line based installation is easier in Drupal or WordPress.
+
+## Testing
+
+The core system is the same across all CMS options, but occasionally there are bugs in the CMS integration code. Our most robust testing is performed on Drupal/CiviCRM which also benefits from the testing inherent in a higher install base, therefore it may take a bit longer to identify/resolve bugs in the WordPress and Joomla integrations but automated testing on these platforms is being continually improved.
```
Comment (mlutfy) [inline docs/planning/cms.md]: I feel that, if we go in this direction, we have to admit that testing is done against Drupal7, the old version fo Drupal that people should not adopt in ~~2017~~ 2019.

The argument about the larger d7 userbase is accurate, but also, we should mention that this is for d7.

File: docs/planning/cms.md (hunk 2 above)
Comment (MikeyMJCO) [reply docs/planning/cms.md]: Okay - I'll take another crack at wording it better and more objectively with some detail :-)

File: docs/planning/cms.md (hunk 1 above)
Comment (mlutfy) [reply docs/planning/cms.md]: I think the wording is good, I would recommend moving this paragraph to the top.

File: docs/planning/cms.md (hunk 3 above)
Comment (mlutfy) [reply docs/planning/cms.md]: To clarify: we do want regular testing on WP, Backdrop and D8. Joomla testing would only be possible if we improve CLI support, which we are pondering on an MIH for that, but might not happen in the short-term. Tim hinted that we could do PR testing against a random CMS at some point, which could be neat.

File: docs/planning/cms.md (hunk 2 above)
Comment (kcristiano) [reply docs/planning/cms.md]: Thanks For working on this.  From a WP perspective-  Roles and Capabilities are 'hidden' from end users, but very easily handled by advanced users or with a pluigin that exposes the system with a UI.  I am not sure we need the 'hidden/advanced' but we surely should not say hidden.   Maybe just advanced?

File: docs/planning/cms.md (hunk 3 above)
Comment (kcristiano) [reply docs/planning/cms.md]: we have also recently added testing for WP and although there are only a couple of unit tests right now, WP is probably the most heavily tested CMS when RC testing is done.   Myu opinion of course, but as I am doing a good deal of the testing and our main test site is WP that is a mitigating factor here.  I also do more testing (RC user side) on Joomla as I know it lacks from automated tests.

File: docs/planning/cms.md [hunk 4]
```diff
@@ -1,61 +1,45 @@
...
+
+The following chapters in this section go into more detail about what is offered for each CMS.
+
+## Users and Contacts
 
+Anonymous site visitors and those with a website login are website **Users**.
+Individuals and organisations with a CiviCRM record are **Contacts**.
```
Comment (totten) [inline docs/planning/cms.md]: I think this section could use more of a transition. The preceding paras are sort of a compare/contrast, but this just seems to give a definition of "Users"/"Contacts".

File: docs/planning/cms.md (hunk 2 above)
Comment (totten) [reply docs/planning/cms.md]: Yeah, just "advanced" is fine. :+1:

(*Either qualifier is accurate from my POV, but including both qualifiers makes it come off as more equivocation than is intended.*)

File: docs/planning/cms.md (hunk 1 above)
Comment (MikeyMJCO) [reply docs/planning/cms.md]: Done - see updated md :-)

File: docs/planning/cms.md (hunk 4 above)
Comment (MikeyMJCO) [reply docs/planning/cms.md]: Removed section - overall wasn't helpful.

**Comment by MikeyMJCO** [comment]
Merging - can be iterated upon but this is an improvement on the current state.

**Pull request merged**